addons/web_editor/models/ir_qweb.py

```diff
@@ -47,10 +47,26 @@ class QWeb(models.AbstractModel):
 
     # compile directives
 
+    def _compile_node(self, el, options):
+        if options.get('is-snippet') == options['template']:
+            # Get the path of the template's content
+            path = options['root'].getpath(el).replace('/t', '')
+            # Only append the template key on template's firt node.
+            # If it already has an data-snippet it is a saved snippet.
+            # Do not override it.
+            size = len(path.split('/'))
+            if size == 2 and 'data-snippet' not in el.attrib:
+                el.attrib['data-snippet'] = options['template']
+
+        body = super()._compile_node(el, options)
+        return body
+
     def _compile_directive_snippet(self, el, options):
         el.set('t-call', el.attrib.pop('t-snippet'))
+        key = el.attrib.get('t-call')
+        el.set('t-call-options', "{'is-snippet': '" + key + "'}")
```

Comment:
Should we remove the app name from the key and only use `s_xxx`? So that if we move a snippet from one module to another it does not get deprecated for a technical reason?